src/metabase/models/pulse.clj

```diff
@@ -85,22 +87,96 @@
   "Fetch a single `Pulse` by its ID value."
   [id]
   {:pre [(integer? id)]}
-  (-> (Pulse id)
-      (hydrate :creator :cards [:channels :recipients])
-      (m/dissoc-in [:details :emails])))
+  (-> (db/select-one Pulse {:where [:and
+                                    [:= :id id]
+                                    [:= :alert_condition nil]]})
+      hydrate-pulse
+      remove-alert-fields))
+
+(defn retrieve-pulse-or-alert
+  "Fetch an alert or pulse by its ID value."
+  [id]
+  {:pre [(integer? id)]}
+  (-> (db/select-one Pulse {:where [:= :id id]})
+      hydrate-pulse))
+
+(defn- pulse->alert
+  "Convert a pulse to an alert"
+  [pulse]
+  (-> pulse
+      (assoc :card (first (:cards pulse)))
+      (dissoc :cards)))
 
+(defn retrieve-alert
+  "Fetch a single alert by its pulse `ID` value."
+  [id]
+  {:pre [(integer? id)]}
+  (-> (db/select-one Pulse {:where [:and
+                                    [:= :id id]
+                                    [:not= :alert_condition nil]]})
+      hydrate-pulse
+      pulse->alert))
+
+(defn retrieve-alerts
+  "Fetch all alerts"
+  []
+  (for [pulse (db/select Pulse, {:where [:not= :alert_condition nil]
+                                 :order-by [[:name :asc]]})]
+
+    (-> pulse
+        hydrate-pulse
+        pulse->alert)))
 
 (defn retrieve-pulses
   "Fetch all `Pulses`."
   []
-  (for [pulse (-> (db/select Pulse, {:order-by [[:name :asc]]})
-                  (hydrate :creator :cards [:channels :recipients]))]
-    (m/dissoc-in pulse [:details :emails])))
-
+  (for [pulse (db/select Pulse, {:where [:= :alert_condition nil]
+                                 :order-by [[:name :asc]]} )]
+    (-> pulse
+        hydrate-pulse
+        remove-alert-fields)))
+
+(defn- query-as [model query]
+  (db/do-post-select model (db/query query)))
+
+(defn retrieve-user-alerts-for-card
+  "Find all alerts for `CARD-ID` that `USER-ID` is set to receive"
+  [card-id user-id]
+  (map (comp pulse->alert hydrate-pulse)
+       (query-as Pulse
+                 {:select [:p.*]
+                  :from   [[Pulse :p]]
+                  :join   [[PulseCard :pc] [:= :p.id :pc.pulse_id]
+                           [PulseChannel :pchan] [:= :pchan.pulse_id :p.id]
+                           [PulseChannelRecipient :pcr] [:= :pchan.id :pcr.pulse_channel_id]]
+                  :where  [:and
+                           [:not= :p.alert_condition nil]
+                           [:= :pc.card_id card-id]
+                           [:= :pcr.user_id user-id]]})))
+
+(defn retrieve-alerts-for-card
+  "Find all alerts for `CARD-ID`, used for admin users"
+  [& card-ids]
+  (map (comp pulse->alert hydrate-pulse)
+       (query-as Pulse
+                 {:select [:p.*]
+                  :from   [[Pulse :p]]
+                  :join   [[PulseCard :pc] [:= :p.id :pc.pulse_id]]
+                  :where  [:and
+                           [:not= :p.alert_condition nil]
+                           [:in :pc.card_id card-ids]]})))
```

Comment:
If you're using `:in` you have to make sure you only do the query in the first place if the seq is non-empty. HoneySQL is dumb and will generate a clause like `WHERE pc.card_id IN ()` which will barf. But you don't need to do the query anyway because nothing is going to come back. So make sure you wrap this whole thing in a `(when (seq card-ids) ...)`